tools/validators/instance_validator/validate/instance_parser.py

```diff
@@ -366,27 +379,39 @@ def _ValidateEntityBlock(self, block: syaml.YAML) -> None:
 
     for key in block.keys():
       if key in self._validated_entities:
-        raise ValueError('Duplicate key {0}'.format(key))
+        raise ValueError("Duplicate key '{0}'".format(key))
       self._ValidateEntityContent(block.get(key))
     self._validated_entities.update(block.data)
 
-  def _ValidateBlock(self, unvalidated_block: str,
+  def _ValidateBlock(self, unvalidated_block: {},
                      validation_fn: Callable[[syaml.YAML], None]) -> None:
     """Validates a yaml-formatted string using the provided function.
 
     Args:
       unvalidated_block: string. A block of unvalidated entities
       validation_fn: a validation function that takes YAML as an argument
     """
+    _msg = "[#{}] Validating '{}'"
+    print(_msg.format(
+      unvalidated_block['source_filenumber'],
```

Comment:
with a class, you can use dot notation (or even better, a getter since the immutable filename and filenumber should be _private)